Flaky test

--- FAIL: TestHooks_OnCodeChangeV2 (0.00s)
    statedb_hooked_test.go:171: error event 2
        have: 0xbb00000000000000000000000000000000000000.code: 0x1326 (0x3c54516221d604e623f358bc95996ca3242aaa109bddabcebda13db9b3f90dcb) -> (0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470) SelfDestruct
        want: 0xaa00000000000000000000000000000000000000.code: 0x1325 (0xa12ae05590de0c93a00bc7ac773c2fdb621e44f814985e72194f921c0050f728) -> (0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470) SelfDestruct
FAIL
FAIL	github.com/ethereum/go-ethereum/core/state	14.796s


see: https://github.com/ethereum/go-ethereum/pull/33628

--- a/core/state/statedb_hooked.go
+++ b/core/state/statedb_hooked.go
@@ -237,7 +237,8 @@ func (s *hookedStateDB) Finalise(deleteEmptyObjects bool) {
 	}
 
 	// Collect all self-destructed addresses first, then sort them to ensure
-	// deterministic ordering when emitting hooks.
+	// that state change hooks will be invoked in deterministic
+	// order when the accounts are deleted below
 	var selfDestructedAddrs []common.Address
 	for addr := range s.inner.journal.dirties {
 		obj := s.inner.stateObjects[addr]

